### PR TITLE
🌱  bump `github.com/docker/docker` from 24.0.7 -> 25.0.0

### LINKED ISSUE #10045: github.com/docker/docker from 24.0.7 to 25.0.0
This bump was attempted by dependabot here: https://github.com/kubernetes-sigs/cluster-api/pull/10044. That attempt failed as some code-level changes are needed in CAPD in order to make this update work.

/help

### PR DESCRIPTION
**Pull request by adityabhatia**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10045

/area dependency

### CODE REVIEW

File: test/infrastructure/container/interface.go [hunk 1]
```diff
@@ -97,8 +99,8 @@ type RunContainerInput struct {
 	// IPFamily is the IP version to use.
 	IPFamily clusterv1.ClusterIPFamily
 	// RestartPolicy to use for the container.
-	// If not set, defaults to "unless-stopped".
-	RestartPolicy string
+	// If not set, defaults to RestartPolicyUnlessStopped.
+	RestartPolicy dockercontainer.RestartPolicyMode
```
Comment (adityabhatia) [inline test/infrastructure/container/interface.go:103]: I could not find any usage for `RestartPolicy` in **public** repositories

**Review by killianmuldoon — COMMENTED** [review]
Thanks for taking this on!

/ok-to-test

**Comment by killianmuldoon** [comment]
BTW - you should really try to get sigs membership - happy to sponsor!

**Comment by sbueringer** [comment]
> BTW - you should really try to get sigs membership - happy to sponsor!

+1

**Comment by chrischdi** [comment]
/lgtm

I would not say that this is a breaking (⚠️ ) change, more seedling 🌱 ?

**Comment by adityabhatia** [comment]
> /lgtm
> 
> I would not say that this is a breaking (⚠️ ) change, more seedling 🌱 ?

Only because this [struct](https://github.com/kubernetes-sigs/cluster-api/pull/10057/files#diff-2d5ac937c8f05844fd0b91b72d2117b53f8fb154bdea0282fb281e8d9545eb39L70) is exported, I marked it as a breaking change, WDYT?

File: test/infrastructure/container/interface.go (hunk 1 above)
Comment (sbueringer) [reply test/infrastructure/container/interface.go:103]: It's fine, see: https://github.com/kubernetes-sigs/cluster-api/blob/f13793dcc1349b5e79f3f7be0737031763d91003/test/README.md#L5

**Comment by sbueringer** [comment]
> > /lgtm
> > I would not say that this is a breaking (⚠️ ) change, more seedling 🌱 ?
> 
> Only because this [struct](https://github.com/kubernetes-sigs/cluster-api/pull/10057/files#diff-2d5ac937c8f05844fd0b91b72d2117b53f8fb154bdea0282fb281e8d9545eb39L70) is exported, I marked it as a breaking change, WDYT?

We have this node that we can do whatever we want basically in this part of our code base: https://github.com/kubernetes-sigs/cluster-api/blob/f13793dcc1349b5e79f3f7be0737031763d91003/test/README.md#L5

But this only means we can break stuff, not how to surface it in release notes.

Basicaly we have the trade-off of "technically it's a breaking change" vs. "it will show up on top of our release notes but 99% of our users don't care and it's easy to adopt to"

I'm fine either way

/lgtm

/assign @fabriziopandini

**Comment by adityabhatia** [comment]
I replac

> > > /lgtm
> > > I would not say that this is a breaking (⚠️ ) change, more seedling 🌱 ?
> > 
> > 
> > Only because this [struct](https://github.com/kubernetes-sigs/cluster-api/pull/10057/files#diff-2d5ac937c8f05844fd0b91b72d2117b53f8fb154bdea0282fb281e8d9545eb39L70) is exported, I marked it as a breaking change, WDYT?
> 
> We have this node that we can do whatever we want basically in this part of our code base:
> 
> https://github.com/kubernetes-sigs/cluster-api/blob/f13793dcc1349b5e79f3f7be0737031763d91003/test/README.md#L5
> 
> But this only means we can break stuff, not how to surface it in release notes.
> 
> Basically we have the trade-off of "technically it's a breaking change" vs. "it will show up on top of our release notes but 99% of our users don't care and it's easy to adopt to"
> 
> I'm fine either way
> 
> /lgtm
> 
> /assign @fabriziopandini

Thanks, going by this it makes I will change it to 🌱 as @chrischdi also suggested.

**Pull request merged**